### ISSUE TITLE
right click on an item to copy command to clipboard

### ISSUE THREAD

**Issue by JoA-MoS** [nrwl/nx-console#1345]
## Description

It would be helpful to be able to copy a command or project task to the clipboard so it could be pasted into the terminal to be modified before executing or use the terminal up arrow shortcut to rerun.

## Motivation

Allow ease of copying, modifying, and rerunning commands directly via the terminal

## Suggested Implementation

in the projects section, I could expand my-app --> build --> production then right-click on production and have a context menu that allows me to copy it to the clipboard.

## Alternate Implementations

It doesn't have to be a context menu could just be a button similar to the open file or play button, maybe the two overlapped pages icon.

**Comment by MaxKless** [comment]
This is a pretty good idea and is somewhere on my list of things to do! 
Would you be interested in contributing this? Should just be a single command and entry into `package.json`. I'd be happy to point you in the right direction.

**Comment by JoA-MoS** [comment]
Yes, I would be happy to contribute. If you could point me in the right direction. I haven't done extensions development before but would enjoy the opportunity.

**Comment by MaxKless** [comment]
That's great to hear @JoA-MoS 
To read about VSCode extension basics, have a look at the docs: https://code.visualstudio.com/api/get-started/extension-anatomy

You can find the code relating to the project view under `libs/vscode/nx-project-view`

To implement this feature, you're going to need to do the following things: 
- Register a command that will be used to do the copying (in `apps/vscode/src/package.json`)
- Register that command to be available in a context menu (make sure that the context menu is only available for items with a context value of `target`)
- write the command implementation using vscode's [Clipboard API](https://code.visualstudio.com/api/references/vscode-api#Clipboard)

This guide should have almost all the info you need: https://code.visualstudio.com/api/extension-guides/command#creating-new-commands

Let me know if you have questions and I look forward to your PR :)

**Comment by MaxKless** [comment]
check out this recent PR that does something similar: https://github.com/nrwl/nx-console/pull/1445

The `package.json` changes should be about the same, you will just have to do some different logic for the copying part.

@JoA-MoS if you don't have time to build this it's totally understandable too! Maybe @dzhavat wants to jump in :P

**Comment by dzhavat** [comment]
I can see a lot of overlap between this request and my PR. Will let @JoA-MoS decide whether he wants to take it. Otherwise I can also chip in. As for the copy functionality, I've also used something similar in one of my extensions. One can also use the Notification API to let the user know that the copy was (un)successful. 

I've used it like this:

https://github.com/dzhavat/git-cheatsheet-inside-vs-code/blob/a6ecacf17791970ad31fc07c77ee658de463d646/src/extension.ts#L26-L31
